ptf/saiacl.py

```diff
@@ -0,0 +1,5852 @@
+        acl_table = sai_thrift_create_acl_table(
+            self.client,
+            acl_stage=table_stage,
+            acl_bind_point_type_list=table_bind_point_type_list,
+            field_src_ip=True)
+
+        # create ACL table entry
+        src_ip = '10.0.0.1'
+        src_ip_mask = '255.255.255.255'
+        src_ip_t = sai_thrift_acl_field_data_t(
+            data=sai_thrift_acl_field_data_data_t(ip4=src_ip),
+            mask=sai_thrift_acl_field_data_mask_t(ip4=src_ip_mask))
+
+        packet_action = sai_thrift_acl_action_data_t(
+            parameter=sai_thrift_acl_action_parameter_t(
+                s32=SAI_PACKET_ACTION_DROP))
+        acl_entry = sai_thrift_create_acl_entry(
+            self.client,
+            table_id=acl_table,
+            priority=10,
+            field_src_ip=src_ip_t,
+            action_packet_action=packet_action)
+
+        # add ACL table group member
+        member1 = sai_thrift_create_acl_table_group_member(
+            self.client,
+            acl_table_group_id=acl_group,
+            acl_table_id=acl_table)
+
+        # create ACL counter
+        acl_counter = sai_thrift_create_acl_counter(
+            self.client, table_id=acl_table)
+
+        # attach ACL counter to ACL entry
+        action_counter_t = sai_thrift_acl_action_data_t(
+            parameter=sai_thrift_acl_action_parameter_t(
+                oid=acl_counter),
+            enable=True)
+        sai_thrift_set_acl_entry_attribute(
+            self.client, acl_entry,
+            action_counter=action_counter_t)
+
+        try:
+            pkt1 = simple_udp_packet(
+                eth_dst=self.mac_port,
+                eth_src=self.mac_lag,
+                ip_src=src_ip,
+                pktlen=100)
+            pkt2 = simple_udp_packet(
+                eth_dst=self.mac_lag,
+                eth_src=self.mac_port,
+                ip_src=src_ip,
+                pktlen=100)
+
+            print("Sending packet without acl table group")
+            print("Sending packet from lag to port")
+            send_packet(self, self.dev_port4, pkt1)
+            verify_packet(self, pkt1, self.dev_port0)
+
+            print("Sending packet from port to lag")
+            send_packet(self, self.dev_port0, pkt2)
+            verify_any_packet_any_port(
+                self, [pkt2, pkt2, pkt2],
+                [self.dev_port4, self.dev_port5, self.dev_port6])
+
+            packets = sai_thrift_get_acl_counter_attribute(
+                self.client, acl_counter, packets=True)
+            self.assertEqual(packets['packets'], 0)
+
+            print("Attach acl table group to port")
+            sai_thrift_set_port_attribute(self.client, self.port0,
+                                          ingress_acl=acl_group)
+            print("Sending packet from port to lag, drop")
+            send_packet(self, self.dev_port0, pkt2)
+            verify_no_other_packets(self)
```

Comment:
Send packet from other ports without ACL binding and check for packet forwarding and no ACL counter incrementing

Reply:
Isn't that already done in lines 150-165?